### PR TITLE
#224 : Base64Decoded Text

### PR DESCRIPTION
**Pull request by memoyil**
Base64Decoded Text

### CODE REVIEW

**Comment by 0crat** [comment]
@memoyil I'm not managing this repo, remove the [webhook](https://github.com/yegor256/cactoos/settings/hooks) or contact me in [Slack](http://www.zerocracy.com) //cc @yegor256

File: src/main/java/org/cactoos/text/Base64DecodedText.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    public Base64DecodedText(final Text text) {
+        this.origin = text;
+    }
+
+    @Override
+    public String asString() throws IOException {
+        final byte[] decoded = Base64.getDecoder().decode(this.origin
+            .asString()
+        );
+        return new String(decoded, "utf-8");
```
Comment (fabriciofx) [inline src/main/java/org/cactoos/text/Base64DecodedText.java]: @memoyil What do you think use `BytesAsText` here?

Comment (memoyil) [inline src/main/java/org/cactoos/text/Base64DecodedText.java]: @fabriciofx I thought the same but at the end of the day the user wants the String form of the text, there is no need to use BytesAsText internally i think.

**Comment by fabriciofx** [comment]
@memoyil I'm sorry, but you opened an issue to this PR?

**Comment by memoyil** [comment]
@fabriciofx  No I didnt, Should I open ?

**Comment by fabriciofx** [comment]
@memoyil Yeah, please. Thanks!

File: src/main/java/org/cactoos/text/Base64DecodedText.java (hunk 1 above)
Comment (fabriciofx) [inline src/main/java/org/cactoos/text/Base64DecodedText.java]: @memoyil I agree, but `ByteAsText` do the service in a more elegant way IMHO. Besides, it's interesting reuse all own classes and maintain the code more declarative possible.

Comment (memoyil) [inline src/main/java/org/cactoos/text/Base64DecodedText.java]: @fabriciofx  I added it. Can you please check

File: src/main/java/org/cactoos/text/Base64DecodedText.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    public Base64DecodedText(final Text text) {
+        this.origin = text;
+    }
+
+    @Override
+    public String asString() throws IOException {
+        final BytesAsText bytes = new BytesAsText(
+            Base64.getDecoder().decode(
+            this.origin.asString()
+        ));
```
Comment (fabriciofx) [inline src/main/java/org/cactoos/text/Base64DecodedText.java]: @memoyil The code is fine, but could you fix the "))" indentation? I'm surprised Qulice let pass it... :)

Comment (memoyil) [inline src/main/java/org/cactoos/text/Base64DecodedText.java]: @fabriciofx  Fixed

File: src/main/java/org/cactoos/text/Base64DecodedText.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+import java.io.IOException;
+import java.util.Base64;
+import org.cactoos.Text;
+
+/**
+ * Base64 Decode the Text.
+ *
+ * @author Mehmet Yildirim (memoyil@gmail.com)
+ * @version $Id$
+ * @since 0.2
```
Comment (yegor256) [inline src/main/java/org/cactoos/text/Base64DecodedText.java]: @memoyil has to be 0.9 here

File: src/main/java/org/cactoos/text/Base64DecodedText.java [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    }
+
+    @Override
+    public String asString() throws IOException {
+        final BytesAsText bytes = new BytesAsText(
+            Base64.getDecoder().decode(
+                this.origin.asString()
+            )
+        );
+        return bytes.asString();
```
Comment (yegor256) [inline src/main/java/org/cactoos/text/Base64DecodedText.java]: @memoyil can't you inline the value and get rid of `bytes` variable?

File: src/main/java/org/cactoos/text/Base64DecodedText.java [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+     * The text.
+     */
+    private final Text origin;
+
+    /**
+     * Ctor.
+     *
+     * @param text The text
+     */
+    public Base64DecodedText(final Text text) {
```
Comment (yegor256) [inline src/main/java/org/cactoos/text/Base64DecodedText.java]: @memoyil let's accept a `String` too

File: src/test/java/org/cactoos/text/Base64DecodedTextTest.java [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+import org.cactoos.TextHasString;
+import org.hamcrest.MatcherAssert;
+import org.junit.Test;
+
+/**
+ * Test case for {@link Base64DecodedText}.
+ *
+ * @author Mehmet Yildirim (memoyil@gmail.com)
+ * @version $Id$
+ * @since 0.2
```
Comment (yegor256) [inline src/test/java/org/cactoos/text/Base64DecodedTextTest.java]: @memoyil must be 0.9

File: src/main/java/org/cactoos/text/Base64DecodedText.java (hunk 5 above)
Comment (memoyil) [inline src/main/java/org/cactoos/text/Base64DecodedText.java]: @yegor256 Done

**Comment by yegor256** [comment]
@rultor try to merge

**Comment by memoyil** [comment]
@yegor256 I fixed the build failure you can try it again

**Comment by yegor256** [comment]
@rultor merge

**Pull request merged**